packages/vscode-extension/src/language/file-system.ts

```diff
@@ -49,10 +49,22 @@ export class VSCodeFileSystemProvider implements FileSystemProvider {
       return undefined;
     }
   }
-  writeFile(_uri: URI, _value: string): Promise<void> {
-    throw new Error("Not supported.");
+  async writeFile(_uri: URI, _value: string): Promise<void> {
+    try {
+      await this._connection.sendRequest(Messages.WriteFile, { _uri, _value });
+    } catch {
+      throw new Error("Couldn't write file");
+    }
```

Comment:
Suggestion: There is no `Messages.WriteFile` implementation on the language client. Also, passing the `URI` object through the connection will result in a serialized `URI` object on the other end - which is rather difficult to work with.
```suggestion
  async writeFile(uri: URI, value: string): Promise<void> {
    try {
      await this._connection.sendRequest(Messages.WriteFile, uri.toString(), value);
    } catch {
      throw new Error("Couldn't write file");
    }
```

Reply:
Hm... ok, gonna work on this. Thanks!